Description: Add --build-date command line flag.
 Allow create reproducible documentation passing a unix time in the command
 line.
Author: Juan Picca <jumapico@gmail.com>
Last-Update: 2015-05-16
Index: texi2html-5.0/T2h_i18n.pm
===================================================================
--- texi2html-5.0.orig/T2h_i18n.pm
+++ texi2html-5.0/T2h_i18n.pm
@@ -315,7 +315,14 @@ sub pretty_date($)
     my $lang = shift;
     my($sec, $min, $hour, $mday, $mon, $year, $wday, $yday, $isdst);
 
-    ($sec, $min, $hour, $mday, $mon, $year, $wday, $yday, $isdst) = localtime(time);
+    if (defined $Texi2HTML::Config::BUILD_DATE)
+    {
+        ($sec, $min, $hour, $mday, $mon, $year, $wday, $yday, $isdst) = gmtime($Texi2HTML::Config::BUILD_DATE);
+    }
+    else
+    {
+        ($sec, $min, $hour, $mday, $mon, $year, $wday, $yday, $isdst) = localtime(time);
+    }
     $year += ($year < 70) ? 2000 : 1900;
     return main::gdt('{month} {day}, {year}', { 'month' => main::gdt($MONTH_NAMES[$mon]),
           'day' => $mday, 'year' => $year });
Index: texi2html-5.0/texi2html.init
===================================================================
--- texi2html-5.0.orig/texi2html.init
+++ texi2html-5.0/texi2html.init
@@ -296,6 +296,10 @@ $L2H = '';
 # output only one file including ToC. It only makes sense when not split
 $MONOLITHIC = 1;
 
+# -build-date
+# Use the given unix time as build date showing UTC timezone when it is used
+$BUILD_DATE = $ENV{'SOURCE_DATE_EPOCH'};
+
 ######################
 # The following options are only relevant if $L2H is set
 #
Index: texi2html-5.0/texi2html.pl
===================================================================
--- texi2html-5.0.orig/texi2html.pl
+++ texi2html-5.0/texi2html.pl
@@ -333,6 +333,7 @@ $EXTERNAL_DIR
 $IGNORE_PREAMBLE_TEXT
 @CSS_FILES
 @CSS_REFS
+$BUILD_DATE
 $INLINE_CONTENTS
 $INLINE_INSERTCOPYING
 $PARAGRAPHINDENT
@@ -3264,6 +3265,13 @@ $T2H_OPTIONS -> {'css-ref'} =
  verbose => 'generate reference to the CSS URL $s'
 };
 
+$T2H_OPTIONS -> {'build-date'} =
+{
+ type => '=i',
+ linkage => \$Texi2HTML::Config::BUILD_DATE,
+ verbose => 'use the given unix time as build date showing UTC timezone when it is used'
+};
+
 $T2H_OPTIONS -> {'transliterate-file-names'} =
 {
  type => '!',
Index: texi2html-5.0/doc/texi2html.texi
===================================================================
--- texi2html-5.0.orig/doc/texi2html.texi
+++ texi2html-5.0/doc/texi2html.texi
@@ -529,6 +529,8 @@ Display a short help and exit.
 Be verbose.
 @item @longopt{version}
 Display version information and exit.
+@item @longopt{build-date=@var{unixtime}}
+Use the given unix time as build date showing UTC timezone when it is used.
 @end table
 
 @c --------------------------------------------------------
Index: texi2html-5.0/doc/texi2html.info
===================================================================
--- texi2html-5.0.orig/doc/texi2html.info
+++ texi2html-5.0/doc/texi2html.info
@@ -439,6 +439,9 @@ Miscellaneous general options:
      
      Display version information and exit.
 
+`--build-date=UNIXTIME'
+     Use the given unix time as build date showing UTC timezone when it is used.
+
 
 File: texi2html.info,  Node: Splitting output,  Next: Output files,  Prev: General options,  Up: Invoking texi2html
 
